### PR TITLE
[wip] update golang, k8s API and argo version

### PR DESCRIPTION
**Pull request by ccfishk**
Fixes https://github.com/keikoproj/addon-manager/issues/93

### CODE REVIEW

**Comment by kevdowney** [comment]
Let's also run BDDs for this, the `make bdd` target should help but you'll need to run the kind setup first `make kind-cluster`

**Comment by ccfishk** [comment]
> Let's create a new issue for Upgrading K8s APIs since you're only addressing the golang and Argo in this PR.

update K8S APIs in this PR also.

File: config/argo/argo.yaml [hunk 1]
```diff
@@ -10,105 +10,105 @@ metadata:
...
+      - ""
+    resources:
+      - persistentvolumeclaims
+    verbs:
+      - create
+      - delete
+      - get
+  - apiGroups:
+      - argoproj.io
+    resources:
```
Comment (kevdowney) [inline config/argo/argo.yaml:52]: This looks like it's missing some of the new RBAC like:
```
      - workflowtasksets
      - workflowtasksets/finalizers
```

See https://github.com/argoproj/argo-workflows/blob/master/manifests/namespace-install/workflow-controller-rbac/workflow-controller-role.yaml

File: go.mod [hunk 2]
```diff
@@ -1,35 +1,69 @@
...
-	github.com/prometheus/common v0.13.0 // indirect
 	github.com/spf13/cobra v1.0.0
-	go.uber.org/zap v1.15.0 // indirect
-	golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a // indirect
 	golang.org/x/net v0.0.0-20201202161906-c7110b5ffcbb
-	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
-	gomodules.xyz/jsonpatch/v2 v2.1.0 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
 	k8s.io/api v0.19.11
 	k8s.io/apiextensions-apiserver v0.19.11
```
Comment (kevdowney) [inline go.mod:16]: When I said k8s API updated it's these dependencies, they should be upgraded to 1.21 at least.
If we want to go to 1.22 it's possible but we need to verify compatibility first.

I don't see you've updated these which you can file a new issue for these.

Comment (ccfishk) [reply go.mod:16]: Right. My original plan is to include the "deprecating old version" in this PR, working on it. 
Thus, revert it WIP.

Comment (ccfishk) [reply go.mod:16]: 92347195c25c4e212f79db24b0a93633ed0675bb

File: config/argo/argo.yaml (hunk 1 above)
Comment (ccfishk) [reply config/argo/argo.yaml:52]: Yes, added here 
https://github.com/keikoproj/addon-manager/pull/95/files#diff-365d1290ca122ba4cd1657bce681b2fb019eb8ee772271d05500f4602dd09eedR55-R56

**Comment by ccfishk** [comment]
This PR is moving to https://github.com/keikoproj/addon-manager/pull/95
Will close as it is.

**Comment by ccfishk** [comment]
close per comment https://github.com/keikoproj/addon-manager/pull/94#issuecomment-1025219286

**Pull request closed without merging**